google_cloud_compute/komand_google_cloud_compute/util/api.py

```diff
@@ -0,0 +1,195 @@
+        request = self.service.disks().createSnapshot(
+            project=self.project_id,
+            zone=zone,
+            disk=disk,
+            body=snapshot_body
+        )
+        return self.execute(request)
+
+    def delete_snapshots(self, snapshot: str):
+        request = self.service.snapshots().delete(
+            project=self.project_id,
+            snapshot=snapshot
+        )
+        return self.execute(request)
+
+    def disk_attach(self, zone: str, instance: str, source: str):
+        attached_disk_body = {
+            "source": source
+        }
+
+        request = self.service.instances().attachDisk(
+            project=self.project_id,
+            zone=zone,
+            instance=instance,
+            body=attached_disk_body
+        )
+        return self.execute(request)
+
+    def disk_detach(self, zone: str, instance: str, device_name: str):
+        request = self.service.instances().detachDisk(
+            project=self.project_id,
+            zone=zone,
+            instance=instance,
+            deviceName=device_name
+        )
+        return self.execute(request)
+
+    def disk_list(self, zone: str, filter: None, max_results: None, order_by: None, page_token: None):
+        if not filter:
+            filter = None
+        if not max_results:
+            max_results = None
+        if not order_by:
+            order_by = None
+        if not page_token:
+            page_token = None
+
+        request = self.service.disks().list(
+            project=self.project_id,
+            zone=zone,
+            filter=filter,
+            maxResults=max_results,
+            orderBy=order_by,
+            pageToken=page_token
+        )
+        return self.execute(request)
+
+    def start_instance(self, zone: str, instance: str):
+        request = self.service.instances().start(
+            project=self.project_id,
+            zone=zone,
+            instance=instance
+        )
+        return self.execute(request)
+
+    def stop_instance(self, zone: str, instance: str):
+        request = self.service.instances().stop(
+            project=self.project_id,
+            zone=zone,
+            instance=instance
+        )
+        return self.execute(request)
+
+    def list_instances(self, zone: str, filter: None, max_results: None, order_by: None, page_token: None):
+        if not filter:
+            filter = None
+        if not max_results:
+            max_results = None
+        if not order_by:
+            order_by = None
+        if not page_token:
+            page_token = None
+
+        request = self.service.instances().list(
+            project=self.project_id,
+            zone=zone,
+            filter=filter,
+            maxResults=max_results,
+            orderBy=order_by,
+            pageToken=page_token
+        )
+        return self.execute(request)
+
+    def list_snapshots(self, filter: None, max_results: None, order_by: None, page_token: None):
```

Comment:
```suggestion
    def list_snapshots(self, filter = None, max_results = None, order_by = None, page_token = None):
```
Did you mean to use default args here? Also another comment about filter shadowing a built-in.